### PR TITLE
pallet-scheduler: fix test

### PR DESCRIPTION
**Pull request by sandreim**
fix https://github.com/paritytech/polkadot-sdk/issues/3921

### CODE REVIEW

File: substrate/frame/scheduler/src/tests.rs [hunk 1]
```diff
@@ -2285,9 +2285,18 @@ fn postponed_named_task_cannot_be_rescheduled() {
...
-		assert!(Preimage::is_requested(&hash));
+
+		// Preimage was not available
+		assert_eq!(
+			System::events().last().unwrap().event,
+			crate::Event::CallUnavailable { task: (4, 0), id: Some(name) }.into()
+		);
+
+		// So it should not be requested.
+		assert!(!Preimage::is_requested(&hash));
```
Comment (bkchr) [inline substrate/frame/scheduler/src/tests.rs:2299]: Some lines below this it is requested again while it wasn't executed. This looks like some potential bug to me. However, I didn't check the logic yet.

**Review by bkchr — COMMENTED** [review]
.

File: substrate/frame/scheduler/src/tests.rs (hunk 1 above)
Comment (sandreim) [reply substrate/frame/scheduler/src/tests.rs:2299]: yeah, but we do call `Preimage::note(call.encode().into()`

Comment (ggwpez) [reply substrate/frame/scheduler/src/tests.rs:2299]: Calling note through the trait instead of the `note_bytes` function does request it indeed.

**Review by ggwpez — APPROVED** [review]

File: substrate/frame/scheduler/src/tests.rs (hunk 1 above)
Comment (sandreim) [reply substrate/frame/scheduler/src/tests.rs:2299]: changed to use the extrinsic.

**Review by bkchr — APPROVED** [review]

**Pull request merged**